src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1093,6 +1093,50 @@ void uct_dc_mlx5_fc_entry_iter_del(uct_dc_mlx5_iface_t *iface, khiter_t it)
     }
 }
 
+static UCS_F_ALWAYS_INLINE void
+uct_dc_mlx5_fc_restore_wnd(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep,
+                           uct_dc_fc_sender_data_t *fc_data)
+{
+    ucs_arbiter_t *waitq;
+    ucs_arbiter_group_t *group;
+    uint8_t pool_index;
+    int16_t cur_wnd;
+    khiter_t it;
+
+    it = kh_get(uct_dc_mlx5_fc_hash, &iface->tx.fc_hash, (uint64_t)ep);
+    if ((fc_data != NULL) &&
+        ((it == kh_end(&iface->tx.fc_hash)) ||
+         (kh_value(&iface->tx.fc_hash, it).seq != fc_data->payload.seq))) {
+        /* Don't process FC_PURE_GRANT further, if an endpoint doesn't exist
+         * anymore, unexpected grant was received on an endpoint or this is not
+         * the same grant sequence number which was expected */
+        return;
+    }
```

Comment:
seems this code should be only in FC_PURE_GRANT handler, and not in the common function

Reply:
refactored